modules/nf-core/cooltools/insulation/main.nf

```diff
@@ -0,0 +1,36 @@
+process COOLTOOLS_INSULATION {
+    tag "${meta.id}"
+    label 'process_medium'
+
+    conda "bioconda::cooltools=0.5.4"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/cooltools:0.5.4--py39hbf8eff0_0' :
+        'biocontainers/cooltools:0.5.4--py39hbf8eff0_0' }"
+
+    input:
+    tuple val(meta), path(cool)
+
+    output:
+    tuple val(meta), path("*tsv"), emit:tsv
+    tuple val(meta), path("*.bw"), optional: true, emit: bigwig
+    path("versions.yml"), emit:versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    cooltools insulation \\
+        -p ${task.cpus} \\
```

Comment:
I think we should only consider the last version ? especially because this module will be designed for the version 0.5.4 (as this is hard coded in the containers directive) ...

@maxulysse, what are the nf-core good practices on that point please ?

Reply:
I would only support the version the module is written for, unless there is a very specific reason why people might want to overwrite it when using a core pipeline. If people are using their own pipelines (or even developing in nf-core) they can use a patch diff or just a local module.